ocaml/xapi/certificates.ml

```diff
@@ -282,12 +285,33 @@ let cert_perms kind =
   debug "%d %d" perm stat.Unix.st_perm ;
   perm
 
+let assert_valid kind ~name ~cert =
```

Comment:
Validation of certificates is usually done in `ocaml/gencert/lib.ml` could you add the function there instead? 

The function `ensure_validity` can be extracted from `validate_certificate` and be reused to generate new error messages, (`ca_certificate_not_valid_yet` and `ca_certificate_expired`)